code/src/pages/CompanyList.js

```diff
@@ -0,0 +1,82 @@
+import React, { useState, useEffect } from 'react';
+import { useParams } from 'react-router-dom';
+
+import Loader from '../components/Loader';
+import Error from '../components/Error';
+import { API_KEY } from '../api.js';
+import CompanyHeader from '../components/CompanyHeader';
+import ButtonBack from '../components/ButtonBack';
+import MovieThumb from '../components/MovieThumb';
+
+const CompanyList = () => {
+  const [isLoading, setLoading] = useState(true);
+  const [company, setCompany] = useState([]);
+  const [companyMovies, setCompanyMovies] = useState([]);
+  const { companyId } = useParams();
+
+  const COMPANY_URL = `https://api.themoviedb.org/3/company/${companyId}?api_key=${API_KEY}`;
+  const MOVIES_URL = `https://api.themoviedb.org/3/discover/movie?api_key=${API_KEY}&language=en-US&sort_by=popularity.desc&include_adult=false&include_video=false&page=1&with_companies=${companyId}`;
+
+  useEffect(() => {
+    fetch(COMPANY_URL)
+      .then(result => {
+        if (result.ok) {
+          return result.json();
+        } else {
+          throw new Error('404');
+        }
+      })
+      .then(json => {
+        console.log(json);
+        setCompany(json);
+        setLoading(false);
+      })
+      .catch(() => {
+        window.location.assign('/error');
```

Comment:
`useHistory().push('/error')` would be more appropriate and clean - we would use react hook